File: VIF_git/GSDS-Qual/ancient/DataStructure/tuple.cpp
#include <iostream>
#include <tuple>
#include <string>
using namespace std;

int main() {
    // Creating a tuple
    tuple<int, double, string> myTuple(42, 3.14, "Hello");

    // Accessing elements
    int firstElement = get<0>(myTuple);
    double secondElement = get<1>(myTuple);
    string thirdElement = get<2>(myTuple);
    cout << "Tuple Elements: " << firstElement << " " << secondElement << " " << thirdElement << endl;

    // Modifying elements
    get<0>(myTuple) = 99;
    get<1>(myTuple) = 2.718;

    cout << "Modified Tuple Elements: " << get<0>(myTuple) << " " << get<1>(myTuple) << " " << get<2>(myTuple) << endl;

    // Tuple size
    size_t tupleSize = tuple_size<decltype(myTuple)>::value;
    cout << "Tuple Size: " << tupleSize << endl;

    // Tuple concatenation
    tuple<int, char> tuple1(10, 'X');
    tuple<double> tuple2(7.5);
    auto concatenatedTuple = tuple_cat(tuple1, tuple2);
    cout << "Concatenated Tuple: " << get<0>(concatenatedTuple) << " " << get<1>(concatenatedTuple) << " " << get<2>(concatenatedTuple) << endl;

    return 0;
}

File: VIF_git/GSDS-Qual/ancient/DataStructure/tuple.py
def main():
    # Creating a tuple
    my_tuple = (42, 3.14, "Hello")

    # Accessing elements
    first_element = my_tuple[0]
    second_element = my_tuple[1]
    third_element = my_tuple[2]

    print(f"Tuple Elements: {first_element} {second_element} {third_element}")

    # Modifying elements (Tuples are immutable, so you can't modify elements directly)
    # Instead, you can create a new tuple with the modified values
    modified_tuple = my_tuple[:2] + (2.718,) + my_tuple[2:]

    print(f"Modified Tuple Elements: {modified_tuple[0]} {modified_tuple[1]} {modified_tuple[2]}")

    # Tuple size
    tuple_size = len(my_tuple)
    print(f"Tuple Size: {tuple_size}")

    # Tuple concatenation
    tuple1 = (10, 'X')
    tuple2 = (7.5,)
    concatenated_tuple = tuple1 + tuple2
    print(f"Concatenated Tuple: {concatenated_tuple[0]} {concatenated_tuple[1]}")

if __name__ == "__main__":
    main()
